packages/loader/container-loader/src/container.ts

```diff
@@ -174,43 +197,22 @@ export async function waitContainerToCatchUp(container: IContainer) {
     }
 
     return new Promise<boolean>((resolve, reject) => {
-        const deltaManager = container.deltaManager;
-
         container.on("closed", reject);
 
-        const waitForOps = () => {
-            assert(container.connectionState !== ConnectionState.Disconnected,
-                0x0cd /* "Container disconnected while waiting for ops!" */);
-            const hasCheckpointSequenceNumber = deltaManager.hasCheckpointSequenceNumber;
-
-            const connectionOpSeqNumber = deltaManager.lastKnownSeqNumber;
-            assert(deltaManager.lastSequenceNumber <= connectionOpSeqNumber,
-                0x266 /* "lastKnownSeqNumber should never be below last processed sequence number" */);
-            if (deltaManager.lastSequenceNumber === connectionOpSeqNumber) {
-                resolve(hasCheckpointSequenceNumber);
-                return;
-            }
-            const callbackOps = (message: ISequencedDocumentMessage) => {
-                if (connectionOpSeqNumber <= message.sequenceNumber) {
-                    resolve(hasCheckpointSequenceNumber);
-                    deltaManager.off("op", callbackOps);
-                }
-            };
-            deltaManager.on("op", callbackOps);
-        };
+        //* I don't fully understand this comment
 
         // We can leverage DeltaManager's "connect" event here and test for ConnectionState.Disconnected
         // But that works only if service provides us checkPointSequenceNumber
         // Our internal testing is based on R11S that does not, but almost all tests connect as "write" and
         // use this function to catch up, so leveraging our own join op as a fence/barrier
         if (container.connectionState === ConnectionState.Connected) {
-            waitForOps();
+            waitForOps(container, resolve);
             return;
         }
 
         const callback = () => {
             container.off(connectedEventName, callback);
-            waitForOps();
+            waitForOps(container, resolve);
         };
         container.on(connectedEventName, callback);
 
```

Comment:
can you please remove eslint comment below?

Reply:
Not going to touch it - `resume` has been deprecated and replaced with `connect`, @scottn12 will be replacing it soon (https://github.com/microsoft/FluidFramework/issues/9397)